test/remote/too_many_authenticated_checks.js

```diff
@@ -0,0 +1,166 @@
+        t.equal(res.statusCode, 200, 'returns a 200 check1')
+        t.equal(obj.block, false, 'not rate limited')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_ONE, ip: TEST_IP, uid: TEST_UID})
+      })
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200 check2')
+        t.equal(obj.block, false, 'not rate limited')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_ONE, ip: TEST_IP, uid: TEST_UID})
+      })
+      // uid should be now blocked
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200 check3')
+        t.equal(obj.block, true, 'uid is rate limited')
+
+        // Delay ~5s for rate limit to go away
+        return Promise.delay(5010)
+      })
+
+      // uid should be now unblocked
+      .then(function(){
+        return client.postAsync('/checkAuthenticated', { action: ACTION_ONE, ip: TEST_IP, uid: TEST_UID})
+      })
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, false, 'is not rate limited after UID_RATE_LIMIT_BAN_DURATION_SECONDS')
+        t.end()
+      })
+      .catch(function(err){
+        t.fail(err)
+        t.end()
+      })
+  }
+)
+
+test(
+  '/checkAuthenticated with different actions',
+  function (t) {
+
+    // Send requests until one gets rate limited
+    return client.postAsync('/checkAuthenticated', { action: ACTION_A, ip: TEST_IP, uid: TEST_UID})
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, false, 'not rate limited on check1, actionA')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_B, ip: TEST_IP, uid: TEST_UID})
+      })
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, false, 'not rate limited on check1, actionB')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_A, ip: TEST_IP, uid: TEST_UID})
+      })
+
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, false, 'not rate limited on check2, actionA')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_A, ip: TEST_IP, uid: TEST_UID})
+      })
+      // uid should be now blocked to action1
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, true, 'uid is actionA rate limited after check3')
+
+        return client.postAsync('/checkAuthenticated', { action: ACTION_B, ip: TEST_IP, uid: TEST_UID})
+      })
+
+      .spread(function(req, res, obj){
+        t.equal(res.statusCode, 200, 'returns a 200')
+        t.equal(obj.block, false, 'not rate limited for actionB after check2')
+
+        // Delay ~5s for rate limit to go away
+        return Promise.delay(5010)
```

Comment:
nit: let's see if we can speed this up and have this be 2 seconds? (so tests run faster)